File: VacancyAccountingSystem/VacancyAccountingSystem/Models/Vacancy.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace VacancyAccountingSystem.Models
{
    public class Vacancy
    {
        public Vacancy()
        {
            VacancySpecialists = new List<VacancySpecialist>();
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string Header { get; set; }

        public string Requirements { get; set; }

        public double? Salary { get; set; }

        public string Offers { get; set; }

        public string About { get; set; }

        [ForeignKey("CompanyFK")]
        public Company Company { get; set; }

        public int? CompanyFK { get; set; }

        public ICollection<VacancySpecialist> VacancySpecialists { get; set; }

    }
}

File: VacancyAccountingSystem/VacancyAccountingSystem/ClientApp/src/app/models/vacancy.ts
import {Company} from './company';

export class Vacancy {
  constructor(about: string, header: string, requirements: string, offers: string, salary: number, company: Company) {

    this.header = header;
    this.requirements = requirements;
    this.offers = offers;
    this.company = company;
    this.about = about;
    this.salary = salary;
  }

  id: number;
  about: string;
  header: string;
  requirements: string;
  offers: string;
  company: Company;
  salary: number;
}
